Public/Src/Utilities/Utilities/Tracing/FormattingEventListener.cs

```diff
@@ -342,26 +342,34 @@ public static string CreateFullMessageString(EventWrittenEventArgs eventData, st
                 }
             }
 
+            ProcessCustomPipDescription(ref full, args.Length, useCustomPipDescription);
+
+            return full;
+        }
+
+        /// <summary>
+        /// Removes BuildXL generated pip description if custom privides its own one
+        /// </summary>
+        protected static void ProcessCustomPipDescription(ref string message, int numberOfArgs, bool useCustomPipDescription)
+        {
             // if args is empty, there is no pip description in the final string

```

Comment:
>if args is empty [](start = 15, length = 16)

nit: I'd change this slightly cause it's not clear what are those "args" the comment is referring to. maybe 'If there are no arguments, ...' ? Cannot say that I like this one either :)